### DISCUSSION TITLE
Can RBAC Role Assignment be done using Azure AD SSO token?

### DISCUSSION THREAD

**Discussion by joshwright10** [langfuse/langfuse#2338] [Support] [answered]
Is it possible to assign users to RBAC roles in Langfuse using the App Role feature in Azure AD?
I see that people are able to do this with next-auth (https://github.com/nextauthjs/next-auth/issues/8070) , but I couldn't see anything in the code that was taking advantage of the `roles` that the Azure AD token is sending.

Is this already available, or would it require a feature request?

**Comment by marcklingen** [comment, marked as answer]
Thanks for the request and pointer to nextauth. Currently this is not supported but we are working on more features related to roles and permissioning

**Reply by joshwright10** [reply]
Thanks @marcklingen. I have opened feature https://github.com/langfuse/langfuse/issues/2345

**Comment by aiakubovich** [comment]
I would be interested in this as well.

We can create groups in Azure (Azure Active Directory Group), and it would be ideally if we could add a group to a specific organization in Langfuse. I'm not sure if this is possible. 

Another simpler way could be to require logging in with some specific Azure AD for a specific Langfuse organization. This way, Azure groups could be added via Azure enterprise application to each such login. For reference you can add Azure groups you created to Azure Enterprise Application. 
So currently Azure AD in Langfuse looks like: 
`Azure Enterprise Application -> Azure AD (App Registrations) -> Langfuse Azure AD login. 
`
My idea to have authentication on organization level: 
```
Azure Enterprise Application #1 -> Azure AD (App Registrations) #1 -> Langfuse Azure AD login for Langfuse Organization #1
Azure Enterprise Application #2 -> Azure AD (App Registrations) #2 -> Langfuse Azure AD login for Langfuse Organization #2
Azure Eenterprise Application #3 -> Azure AD (App Registrations) #3 -> Langfuse Azure AD login for Langfuse Organization #3
```
